src/classes/frModel.cls

```diff
@@ -0,0 +1,71 @@
+public abstract class frModel {
+	protected abstract SObject getObject();
+    public abstract List<frMapping__c> getMappings();
+
+    public virtual void populateFromRequest(Map<String, Object> request) {
+    	Map<String, Schema.SObjectField> fields = frSchemaUtil.getFields(getObject().getSObjectType().getDescribe().getName());
+
+        Map<String, frMapping__c> frNameToSfName = new Map<String, frMapping__c>();
+        for(frMapping__c mapping : getMappings()) {
+            frNameToSfName.put(mapping.fr_Name__c, mapping);
```

Comment:
Yeah this is a good catch.  I'll take another look at this entire method because this part especially doesn't jive well with constant mappings, since those ones dont need a funraise field